core/src/io/anuke/mindustry/ui/fragments/PlacementFragment.java

```diff
@@ -98,18 +132,32 @@ boolean gridUpdate(InputHandler input){
                 i++;
             }
             return true;
-        }else{ //select block
+        } else if (Core.input.keyDown(Binding.gridMode)) { //select block
             int i = 0;
             Array<Block> recipes = getByCategory(currentCategory);
             for(KeyCode key : inputGrid){
                 if(Core.input.keyDown(key))
                     input.block = (i < recipes.size && unlocked(recipes.get(i))) ? recipes.get(i) : null;
                 i++;
             }
+            return true;
         }
+
+        for (int i=0; i<categoryKey.length; i++) {
```

Comment:
Unnecessary spaces around parentheses, should be `for(){` / `}else{`
This occurs throughout the PR.

Reply:
Fixed.